### DISCUSSION TITLE
Are the Maintainers Happy with The Codebase Being Written in Bash?

### DISCUSSION THREAD

**Discussion by eeriemyxi** [ivan-hc/AM#1249] [General]
Hi, I was checking out the source code and found out that it's made up of shell scripts that depend on external tools like `grep`, `wget`, etc.,  all the way deep. Combine that with the fact that it's written in Bash. I am wondering whether the lead maintainers are happy with this decision. Are you guys happy with having to maintain so many shell scripts? I would ideally expect this kind of tool where syntatic errors are crucial to be avoided to be written in a general programming language with at least some bit of type-safety.

**Comment by Samueru-sama** [comment]
The only thing I'm not happy is that AM depends on bash instead of being fully posix shell. 

What would be the alternative to having to maintain many shell scripts? Like for example in archlinux there are PKGBUILD scripts which are similar and ours aren't as complicated since we don't do dependency resolution and similar. 

>I would ideally expect this kind of tool where syntatic errors are crucial to be avoided to be written in a general programming language with at least some bit of type-safety.

You mean something like this? https://github.com/pkgforge/soar

**Comment by ivan-hc** [comment]
Hi, bash/shell is the only language I know. If I didn't make everything POSIX it's because some commands only work in bash.
